test/blackbox-tests/test-cases/formatting/run.t

```diff
@@ -42,25 +42,30 @@ Configuration files are taken into account for this action:
 
   $ touch enabled/.ocamlformat
   $ dune build --display short @fmt
+         refmt enabled/.formatted/reason_file.re
   File "enabled/reason_file.re", line 1, characters 0-0:
   Files _build/default/enabled/reason_file.re and _build/default/enabled/.formatted/reason_file.re differ.
+         refmt enabled/.formatted/reason_file.rei
   File "enabled/reason_file.rei", line 1, characters 0-0:
   Files _build/default/enabled/reason_file.rei and _build/default/enabled/.formatted/reason_file.rei differ.
-  File "enabled/dune", line 1, characters 0-0:
-  Files _build/default/enabled/dune and _build/default/enabled/.formatted/dune differ.
-  File "enabled/subdir/dune", line 1, characters 0-0:
-  Files _build/default/enabled/subdir/dune and _build/default/enabled/subdir/.formatted/dune differ.
-  File "partial/a.ml", line 1, characters 0-0:
-  Files _build/default/partial/a.ml and _build/default/partial/.formatted/a.ml differ.
    ocamlformat enabled/.formatted/ocaml_file.mli
   File "enabled/ocaml_file.mli", line 1, characters 0-0:
   Files _build/default/enabled/ocaml_file.mli and _build/default/enabled/.formatted/ocaml_file.mli differ.
    ocamlformat enabled/.formatted/ocaml_file.ml
   File "enabled/ocaml_file.ml", line 1, characters 0-0:
   Files _build/default/enabled/ocaml_file.ml and _build/default/enabled/.formatted/ocaml_file.ml differ.
+          dune enabled/.formatted/dune
+  File "enabled/dune", line 1, characters 0-0:
+  Files _build/default/enabled/dune and _build/default/enabled/.formatted/dune differ.
+          dune enabled/subdir/.formatted/dune
+  File "enabled/subdir/dune", line 1, characters 0-0:
+  Files _build/default/enabled/subdir/dune and _build/default/enabled/subdir/.formatted/dune differ.
    ocamlformat enabled/subdir/.formatted/lib.ml
   File "enabled/subdir/lib.ml", line 1, characters 0-0:
   Files _build/default/enabled/subdir/lib.ml and _build/default/enabled/subdir/.formatted/lib.ml differ.
+   ocamlformat partial/.formatted/a.ml
```

Comment:
it seems that the dependencies are less precise than before, it's reformatting in partial/ and dune files as well above. do you know why that's the case?

Reply:
This part happens just after `enabled/.ocamlformat` was touched. In contrast, `partial/a.ml` should only be impacted to changes to `.ocamlformat` and `partial/.ocamlformat`. Does that make sense?

Reply:
Correct me if I'm wrong, but `@fmt` should apply to all the subdirs. In the last run, the output of `partial/a.ml`, so to me it makes sense that dune should complain that it hasn't been formatted when building the `@fmt` alias.

Reply:
Yes, `@fmt` should apply to all subdirs. But here it's a dependency issue: neither `partial/a.ml` nor configuration files that affect its formatting have changed, so it does not need to be reformatted.

Reply:
I see what you mean, but I think this behavior is not desirable.

Here's what I expect after running `$ dune build @dir/fmt`.

All files in `dir` are either well formatted or dune will suggest promotions to correct that.

So to me, it seems like it's only safe not to run `ocamlformat` only if:

* The files have not changed
* We know they were well formatted from the previous run.

Note that this is similar to the way `$ dune runtest` for example. Where if a test fails, it will keep failing on subsequent runs.

However, I'm still very much puzzled by the difference in behavior here.

Reply:
Ah! I missed that there are many ways to fail, but only one way to succeed :) In that case, I agree, you can setup this behaviour.

Reply:
Okay. I will fix it a separate PR as I think it deserves its own change entry.

Reply:
:+1: